File: vol2/ch17/LeetCode167-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n=len(numbers)
        for i in range(0,n):
            low,high=i+1,n-1
            while low<=high:
                mid=(low+high)//2
                if numbers[mid]==target-numbers[i]:
                    return [i+1,mid+1]
                elif target-numbers[mid]<numbers[i]:
                    high=mid-1
                else:
                    low=mid+1
        return [0,0]

File: vol2/ch17/LeetCode167-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers,int target) {
        int n=numbers.size();
        for(int i=0;i<n;i++) {
            int low=i+1,high=n-1;
            while(low<=high) {
                int mid=(low+high)/2;
                if(numbers[mid]==target-numbers[i])
                    return {i+1,mid+1};
                else if(target-numbers[mid]<numbers[i])
                    high=mid-1;
                else
                    low=mid+1;
            }
        }
        return {0,0};							//ÎªÁËÓï·¨ 
    }
};

